### PR TITLE
Stabilize `cfg_target_abi`

### LINKED ISSUE #78791: target_env should not include ABI information
There are a few tier-3 targets that have ABI descriptors in their `target_env`.

```
compiler/rustc_target/src/spec/armv6_unknown_freebsd.rs:13:        target_env: "gnueabihf".to_string(),
compiler/rustc_target/src/spec/armv6_unknown_netbsd_eabihf.rs:14:        target_env: "eabihf".to_string(),
compiler/rustc_target/src/spec/armv7_unknown_freebsd.rs:13:        target_env: "gnueabihf".to_string(),
compiler/rustc_target/src/spec/armv7_unknown_netbsd_eabihf.rs:13:        target_env: "eabihf".to_string(),
```

The precedent from other targets is that these should be just "gnu" or an empty string. This came up in the RFC for `target_abi`, https://github.com/rust-lang/rfcs/pull/2992#issuecomment-718099185, where I'd expect these targets will have `target_abi: "eabihf"`.

### LINKED ISSUE #80970: Tracking Issue for cfg-target-abi
This is a tracking issue for the RFC "Add `target_abi` configuration" (rust-lang/rfcs#2992).
The feature gate for the issue is `#![feature(cfg_target_abi)]`.

### About tracking issues

Tracking issues are used to record the overall progress of implementation.
They are also used as hubs connecting to other relevant issues, e.g., bugs or open design questions.
A tracking issue is however *not* meant for large scale discussion, questions, or bug reports about a feature.
Instead, open a dedicated issue for the specific matter and add the relevant feature gate label.

### Steps

- [x] Implement the RFC (cc @rust-lang/lang -- can anyone write up mentoring
      instructions?)
- [x] Fix up tier 3 targets that have target ABIs embedded in their target env (#78791)
- [ ] Adjust documentation ([see instructions on rustc-dev-guide][doc-guide])
- [ ] Stabilization PR ([see instructions on rustc-dev-guide][stabilization-guide])

[stabilization-guide]: https://rustc-dev-guide.rust-lang.org/stabilization_guide.html#stabilization-pr
[doc-guide]: https://rustc-dev-guide.rust-lang.org/stabilization_guide.html#documentation-prs

### PR DESCRIPTION
**Pull request by ChrisDenton**
This stabilizes the `cfg` option called `target_abi`:

```rust
#[cfg(target_abi = "eabihf")]
```

Tracking issue: #80970

fixes #78791
resolves #80970

### CODE REVIEW

**Comment by Nilstrieb** [comment]
Removing nomination as the tracking issue is nominated already (cc @traviscross)

**Comment by wesleywiser** [comment]
Just a note since we came across this in the compiler team triage meeting, we believe this is waiting for the lang team to respond in #80970. If this is waiting on some action from the compiler team, please let us know!

**Comment by ChrisDenton** [comment]
This has been getting a lot attention recently. I probably jumped the gun on posting the stabilization PR before the lang team has even started discussing stabilization. I can close if that would be help get it off everyone's plate?

**Comment by wesleywiser** [comment]
It's not a big deal, we just happened upon it while looking at `T-compiler`+`S-waiting-on-team` issues 🙂

**Comment by madsmtm** [comment]
Suggestion: Add "fixes https://github.com/rust-lang/rust/pull/119590" to the issue description.

**Comment by ChrisDenton** [comment]
The [fcp has completed](https://github.com/rust-lang/rust/issues/80970#issuecomment-1962412608), this is now ready for review.

File: compiler/rustc_target/src/spec/targets/armv6_unknown_freebsd.rs [hunk 1]
```diff
@@ -8,8 +8,7 @@ pub fn target() -> Target {
         arch: "arm".into(),
         options: TargetOptions {
             abi: "eabihf".into(),
-            // FIXME: change env to "gnu" when cfg_target_abi becomes stable
-            env: "gnueabihf".into(),
+            env: "gnu".into(),
```
Comment (Nilstrieb) [inline compiler/rustc_target/src/spec/targets/armv6_unknown_freebsd.rs:11]: Are there cfgs in std that rely on this? Or is this only used in third party code? It also makes sense to ping the target maintainers (if they exist) here to notify them.

Comment (ChrisDenton) [reply compiler/rustc_target/src/spec/targets/armv6_unknown_freebsd.rs:11]: None in std. I did a ripgrep on all of library/ and the only thing that came up was stdarch where it was only used as part of the full target name.

**Comment by ChrisDenton** [comment]
cc @he32, stabilizing this affects at least one netbsd target (`armv6-unknown-netbsd-eabihf`), though hopefully not in a bad way.

**Review by Billyum50 — APPROVED** [review]

**Review by vishalverma8200 — COMMENTED** [review]
One problem

**Pull request merged**